### ISSUE TITLE
Horizon returns 500 for /paths endpoint

### ISSUE THREAD

**Issue by mlsteele** [stellar/go#1496]
This request cause horizon to return 500 "Internal Server Error". On horizon.stellar.org and our own horizon.

https://horizon.stellar.org/paths?source_account=GBGSPJUIK5W5S7QNSFLL5YJTLTPMWD6Q3GHTSLLB3Q6GMBWNC7ST23UA&destination_account=GC3MRN2NVLDDOLKWD46HZKD4NAGDIXRZUKR227NDOT2TFX4ZYZE2N74X&destination_asset_type=credit_alphanum4&destination_asset_code=KEYZ&destination_asset_issuer=GDVDSIDVX63LKSUFTO5NQ3YHPDDAYUSEPOJLFXVWCVCLEKMOERGHZ7DR&destination_amount=1

Changing any of the inputs is pretty likely to result in a 200. In particular, if `destination_amount` is <= 1 then 500 is returned. But if `destination_amount` > 1 e.g. `1.001` then it returns 200.

Some logs:
```
Jul  9 20:39:47 ip-10-0-7-223 stellar-horizon[15871]: time="2019-07-09T20:39:47.064Z" level=info msg="Starting request" app_name=undefined app_version=undefined client_name=undefined client_version=undefined forwarded_ip= host="10.0.7.223:8000" ip=10.0.10.90 ip_port="10.0.10.90:42116" method=GET path="/paths?source_account=GBGSPJUIK5W5S7QNSFLL5YJTLTPMWD6Q3GHTSLLB3Q6GMBWNC7ST23UA&destination_account=GCZATXK4ZQAXOQ26EP6QVZJXJZFCQR5NYNY7AC267HV72YDPLVGVG2ZH&destination_asset_type=credit_alphanum4&destination_asset_code=KEYZ&destination_asset_issuer=GDVDSIDVX63LKSUFTO5NQ3YHPDDAYUSEPOJLFXVWCVCLEKMOERGHZ7DR&destination_amount=1" pid=15871 req=ip-10-0-7-223/QTg3M8hGmp-4020149 streaming=false
Jul  9 20:39:47 ip-10-0-7-223 stellar-horizon[15871]: time="2019-07-09T20:39:47.065Z" level=info msg="Starting pathfind" destination_amount=10000000 destination_asset=credit_alphanum4/KEYZ/GDVDSIDVX63LKSUFTO5NQ3YHPDDAYUSEPOJLFXVWCVCLEKMOERGHZ7DR pid=15871 source_assets="[credit_alphanum4/BTC/GAUTUYY2THLF7SGITDFMXJVYH3LHDSMGEAKSBU267M2K7A3W543CKUEF credit_alphanum4/BTC/GBSTRH4QOTWNSVA6E4HFERETX4ZLSR3CIUBLK7AXYII277PFJC4BBYOG credit_alphanum4/USD/GBSTRUSD7IRX73RQZBL3RQUH6KS3O4NYFY3QCALDLZD77XMZOPWAVTUK credit_alphanum4/HUG/GD5T6IPRNCKFOHQWT264YPKOZAWUMMZOLZBJ6BNQMUGPWGRLBK3U7ZNP credit_alphanum4/TOAD/GDGHHBNYTCH45RQYQE4Z6F2E3JMMBFHFGMF3DBZ3JODDNT3UWEDP5AT4 credit_alphanum4/USD/GDUKMGUGDZQK6YHYA5Z6AY2G4XDSZPSZ3SW5UN3ARVMO6QSRDWP5YLEX credit_alphanum4/KEYZ/GDVDSIDVX63LKSUFTO5NQ3YHPDDAYUSEPOJLFXVWCVCLEKMOERGHZ7DR native]"
```

```
Jul  9 20:39:47 ip-10-0-7-223 stellar-horizon[15871]: time="2019-07-09T20:39:47.135Z" level=info msg="Finished pathfind" err="adding these two values will cause an integer overflow: 6562737123349420122, 4612630176791341640" found=2 pid=15871
Jul  9 20:39:47 ip-10-0-7-223 stellar-horizon[15871]: time="2019-07-09T20:39:47.135Z" level=error msg="adding these two values will cause an integer overflow: 6562737123349420122, 4612630176791341640" pid=15871 req=ip-10-0-7-223/QTg3M8hGmp-4020149 stack=unknown
Jul  9 20:39:47 ip-10-0-7-223 stellar-horizon[15871]: time="2019-07-09T20:39:47.135Z" level=info msg="Finished request" app_name=undefined app_version=undefined bytes=444 client_name=undefined client_version=undefined duration=0.070471857 forwarded_ip= host="10.0.7.223:8000" ip=10.0.10.90 ip_port="10.0.10.90:42116" method=GET path="/paths?source_account=GBGSPJUIK5W5S7QNSFLL5YJTLTPMWD6Q3GHTSLLB3Q6GMBWNC7ST23UA&destination_account=GCZATXK4ZQAXOQ26EP6QVZJXJZFCQR5NYNY7AC267HV72YDPLVGVG2ZH&destination_asset_type=credit_alphanum4&destination_asset_code=KEYZ&destination_asset_issuer=GDVDSIDVX63LKSUFTO5NQ3YHPDDAYUSEPOJLFXVWCVCLEKMOERGHZ7DR&destination_amount=1" pid=15871 req=ip-10-0-7-223/QTg3M8hGmp-4020149 route=/paths status=500 streaming=false
```

**Comment by marcinx** [comment]
Observing the same.

Works fine with amount 10:
https://horizon.stellar.org/paths?source_account=GASKYWPPQ2VSO6KNIPIRVXMSSDGZLYZQ67CDTLVXOXYDG26SPZ66EDCQ&destination_account=GBFNWESKGTN2GDED3ZF5OI4XKFKCFLMSAU5UUU7YCFNMAYCJLJLGUDCQ&destination_asset_type=credit_alphanum4&destination_asset_code=USD&destination_asset_issuer=GDUKMGUGDZQK6YHYA5Z6AY2G4XDSZPSZ3SW5UN3ARVMO6QSRDWP5YLEX&destination_amount=10

Fails with amount 100:
https://horizon.stellar.org/paths?source_account=GASKYWPPQ2VSO6KNIPIRVXMSSDGZLYZQ67CDTLVXOXYDG26SPZ66EDCQ&destination_account=GBFNWESKGTN2GDED3ZF5OI4XKFKCFLMSAU5UUU7YCFNMAYCJLJLGUDCQ&destination_asset_type=credit_alphanum4&destination_asset_code=USD&destination_asset_issuer=GDUKMGUGDZQK6YHYA5Z6AY2G4XDSZPSZ3SW5UN3ARVMO6QSRDWP5YLEX&destination_amount=100

Can't wait for in-memory /paths.

**Comment by marcinx** [comment]
@tomquisel @bartekn Is there any way to prioritize/speed up the /paths refactoring?

Currently Horizon /paths is badly broken. It returns 500 errors too frequently (see above). Astrograph's pathfinder has some bugs and Johansten's has been out of commission for a few months. How does everyone calculate best paths at the moment? I'd rather not have to write our own pathfinder, especially with the refactored Horizon /paths just around the corner (hopefully).

**Comment by tomquisel** [comment]
Yeah, this is a bummer. It's very high priority! In fact, we're shooting to have a release candidate with the new path finder this month. @ire-and-curses heads up about the importance of this.

**Comment by marcinx** [comment]
I saw the WIP check-ins. Great news. Thanks for the update. 

Maybe this would be a good opportunity to retire the mandatory `destination_account` parameter as it's not being used by the path finder?

**Comment by johansten** [comment]
Stargazer is all path-payments, so it is completely out of commission. Just saying.

**Comment by patrickxb** [comment]
Any updates on this one?

**Comment by tomquisel** [comment]
We should be getting a release out this week! I'm not sure if new pathfinding will be turned on by default though. @ire-and-curses  ?

**Comment by patrickxb** [comment]
ok, thanks.

**Comment by ire-and-curses** [comment]
This is targeted for Horizon 0.20 (next release). The work is tied closely to the new ingestion system so needs careful testing. We're aiming for a release candidate by the end of this week, and then we'll spend a week monitoring and testing. Barring any significant issues, the actual release will likely be early in the week after (i.e. the week of the 19th August).

I expect we'll turn it on by default since it should be much much better than the current situation.

**Comment by johansten** [comment]
> I expect we'll turn it on by default since it should be much much better than the current situation.

Not that hard considering it's not working at all right now, and hasn't for months(?)

**Comment by bartekn** [comment]
We believe this has been fixed in `0.20.0` release (and `0.20.1` that contains some performance fixes). Horizon `0.20.1rc1` is currently deployed to https://horizon.stellar.org. Please check it out.
